src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1285,6 +1298,156 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
             set_api_call_source(df, f"DataFrameReader.{format.lower()}")
         return df
 
+    @private_preview(version="1.38.0")
+    @publicapi
+    def jdbc(
+        self,
+        url: str,
+        udtf_configs: dict,
+        *,
+        properties: Optional[dict] = None,
+        table: Optional[str] = None,
+        query: Optional[str] = None,
+        column: Optional[str] = None,
+        lower_bound: Optional[Union[str, int]] = None,
+        upper_bound: Optional[Union[str, int]] = None,
+        num_partitions: Optional[int] = None,
+        query_timeout: Optional[int] = 0,
+        fetch_size: Optional[int] = 0,
+        custom_schema: Optional[Union[str, StructType]] = None,
+        predicates: Optional[List[str]] = None,
+        session_init_statement: Optional[List[str]] = None,
+        _emit_ast: bool = True,
+    ) -> DataFrame:
+        """
```

Comment:
we should call out the credential part
e.g., password in url will be ignored, eai with secret is required